gpflow/base.py

```diff
@@ -103,35 +103,32 @@ def __init__(
         therefore we need a positive constraint and it is natural to use constrained values.
         A prior can be imposed either on the constrained version (default) or on the unconstrained version of the parameter.
         """
-        super().__init__()
+        if transform is None:
+            transform = tfp.bijectors.Identity()
+
+        value = _cast_to_dtype(value, dtype)
+        value_dtype = value.dtype
```

Comment:
variable `value_dtype` is not used anywhere - maybe remove? Are we certain that `dtype` == `value_dtype` as a result of the `_cast_to_dtype`?